core/Test/Tasty/Run.hs

```diff
@@ -219,14 +221,21 @@ type Tr = Traversal
 
 -- | Exceptions related to dependencies between tests.
 data DependencyException
-  = DependencyLoop
-    -- ^ Test dependencies form a loop. In other words, test A cannot start
+  = DependencyLoop [[Path]]
+    -- ^ Test dependencies form cycles. In other words, test A cannot start
     -- until test B finishes, and test B cannot start until test
-    -- A finishes.
+    -- A finishes. Field lists detected cycles.
   deriving (Typeable)
 
 instance Show DependencyException where
-  show DependencyLoop = "Test dependencies form a loop."
+  show (DependencyLoop css) = "Test dependencies have cycles:" ++ showCycles css
+    where
+      showCycles = intercalate "\n- " . map showCycle
```

Comment:
We need `"\n- "` before the first line as well, otherwise the output looks like
```
mod-bench: Test dependencies have cycles:All.Power.Data.Mod, All.Power.Data.Mod
```

Reply:
Good catch, thanks!